### PR TITLE
Npm for all packages

### PR DESCRIPTION
**Pull request by andrasbacsai**
## Before requesting a review

- [ ] if the PR is WIP - use draft mode
- [x] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")
- [ ] what kind of review is needed?
  - [x] conceptual
  - [ ] detailed
  - [ ] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [x] tested locally and on preview environment
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/tsconfig.json [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
-  // Extends can't use node's resolution rn https://github.com/dividab/tsconfig-paths/issues/209#issuecomment-1177261091
-  "extends": "../../node_modules/tsconfig/remix.json",
+  "extends": "../../node_modules/@webstudio-is/tsconfig/remix.json",
   "include": ["remix.env.d.ts", "**/*.ts", "**/*.tsx"],
   "compilerOptions": {
     "paths": {
       "~/*": ["app/*"]
     },
-    "baseUrl": "."
+    "baseUrl": ".",
```
Comment (SaraVieira) [inline apps/designer/tsconfig.json:8]: Why were these lines added?

File: .github/workflows/eslint-config-custom-publish.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: Eslint Config Publish
```
Comment (kof) [inline .github/workflows/eslint-config-custom-publish.yml:1]: looks like something we will need in every published package, can we reuse it somehow?

File: packages/asset-uploader/jest.config.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // eslint-disable-next-line
-const base = require("jest-config/jest.config.js");
+const base = require("@webstudio-is/jest-config/jest.config.js");
```
Comment (kof) [inline packages/asset-uploader/jest.config.js:2]: shouldn't this be, couldrename jest.config.js to index.js
```suggestion
const base = require("@webstudio-is/jest-config");
```

File: apps/designer/tsconfig.json (hunk 1 above)
Comment (andrasbacsai) [reply apps/designer/tsconfig.json:8]: By mistake.

File: packages/asset-uploader/jest.config.js (hunk 3 above)
Comment (andrasbacsai) [reply packages/asset-uploader/jest.config.js]: Fixed it.

File: .github/workflows/eslint-config-custom-publish.yml (hunk 2 above)
Comment (andrasbacsai) [reply .github/workflows/eslint-config-custom-publish.yml:1]: The name? Why we would like to reuse the name?

Comment (SaraVieira) [reply .github/workflows/eslint-config-custom-publish.yml:1]: He means the whole file, most of it is the same in most GitHub actions

**Comment by kof** [comment]
@andrasbacsai is this waiting for a review? does it have remaining tasks? Can't say from description

**Comment by andrasbacsai** [comment]
No, it can be merged.

File: .github/workflows/eslint-config-custom-publish.yml (hunk 2 above)
Comment (kof) [reply .github/workflows/eslint-config-custom-publish.yml:1]: what about this? do you want us to maintain this config for each package? this sounds like a bad idea

Comment (kof) [reply .github/workflows/eslint-config-custom-publish.yml:1]: seems like its possibe to use `uses` and `using`? https://stackoverflow.com/questions/65532726/github-actions-extend-existing-job

**Comment by kof** [comment]
@andrasbacsai is there anything that we need to document for us as a user?

File: .github/workflows/eslint-config-custom-publish.yml (hunk 2 above)
Comment (andrasbacsai) [reply .github/workflows/eslint-config-custom-publish.yml]: Figured out how it can be used.

**Comment by andrasbacsai** [comment]
> @andrasbacsai is there anything that we need to document for us as a user?

Nothing. If someone bumps the version inside a `package.json` under `packages/` a new version will be published to NPM automatically.

I also added icons (`@webstudio-is/icons`) to the [list](https://www.npmjs.com/package/@webstudio-is/icons).

**Comment by kof** [comment]
- [ ] please bring in design system package which was merged

**Comment by kof** [comment]
Can those package specific configs be in the corresponding package?

**Comment by andrasbacsai** [comment]
> Can those package specific configs be in the corresponding package?

Do you mean the workflow files like `.github/workflows/eslint-config-custom.yml`?

**Review by kof — APPROVED** [review]
folow up issue https://github.com/webstudio-is/webstudio-designer/issues/204
